_posts/2019-02-20-lecture-notes-template.md

```diff
@@ -0,0 +1,361 @@
+
+This motivates approximate inference: we trade off exact computation for reasonable computational
+work. We will cover two major families of methods in this lecture and next:
+Variational Inference and Markov Chain Monte Carlo.
+
+
+## Variational Inference
+
+Variational inference is a technique that allows one to convert an inference problem into an
+optimization problem. We want to maximize the data likelihood, but as we already discussed, 
+that itself is too hard to handle. Thus, we begin by lower-bounding the data likelihood
+with a term known as the "free-energy".
+
+
+We start by introducing another distribution , $Q_{\theta}$ which is a distribution on the 
+same set of random variables. The goal, eventually, is to have $Q_{\theta}$ be a distribution
+that is easier to work with than $p$ but is sufficiently "close" to serve as a reasonable 
+surrogate in calculations. 
+
+Using this, we want to maximize the lower bound for the log-likelihood:
+
+$$log(p(x)) = 
+KL(q_{\theta}(z | x) || p_{\theta}(z | x)) + \int_z q_{\theta}(z | x)log\frac{p_{\theta}(x, z)}{q_{\theta}(z | x)}dz 
+\ge \int_z q_{\theta}(z | x)log\frac{p_{\theta}(x, z)}{q_{\theta}(z | x)}dz
+:= \mathscr{L}(\theta, \phi ; x)$$
+
+Equivalently, we can minimize the aforementioned "free-energy" of the system:
+
+
+$$F(\theta, \phi ; x) = -log(p(x)) + KL(q_{\theta}(z | x) || p_{\theta}(z | x))$$
+
+Intuitively, the connection between $$\mathscr{L}(\theta, \phi ; x)$$ and $$F(\theta, \phi ; x)$$ is that the $$KL$$ divergence measures the gap between the lowerbound on the likelyhood ($$\mathscr{L}$$) and the real likliehood ($$log(p(x))$$) - both the minimization and the maximization noted above try to close that gap.
+
+
+We call $$\mathscr{L}(\theta, \phi ; x)$$ above the variational lower bound.
+Often it is written as $$\mathscr{L}(\theta, \phi ; x) = log(p(x)) - KL(q_{\theta}(z | x) || p_{\theta}(z | x))$$, which simply is $$\mathscr{L}(\theta, \phi ; x) = -F(\theta, \phi ; x)$$ .
+
+
+
+## Mean-Field Assumption (In Topic Models)
+
+Recall the form of the true posterior :
+
+$$p(\beta, \theta, z | w) = \frac{p(\beta, \theta, z, w)}{p(w)}$$
+
+Suppose that in q (our approximation to p) we could break dependancies in the jiont by assuming
+a so-called "fully-factorized" distribution is followed, i.e.:
+
+$$q(\beta, \theta, z) = \Pi_{k}q(\beta_k)\Pi_{d}q(\theta_d)\Pi_{n}q(z_{d_n})$$
+
+Notice that in this fully-factored model, each factor is a term of a distribution - 
+this is unlike the factorization we say over Bayes Nets before, where conditional 
+distributions appear. As such, if we need to answer a query of form $$q(z_d)$$, 
+marginalizing across $$q(\beta, \theta, z)$$ is trivial since we now that moving the
+sums into the product, $$\sum_{\beta, \theta}\Pi_{k}q(\beta_k)\Pi_{d}q(\theta_d) = 1$$.
+
+In general with variational methods, the true posterior might not exist in the 
+family of $$q$$ we consider - we buy tractability with the cost of approximation error.
+
```

Comment:
please go through and proof-read one more time!

Reply:
I added some clarification in regards of not needing to marginalize on a Q with the given form. I hope that this addresses the concern you had; I presume the wording provided previously suggested to you that marginalization would still be necessary.